lib/elixir/lib/file.ex

```diff
@@ -361,6 +361,41 @@ defmodule File do
   end
 
   @doc """
+  Reads the symbolic link at `path`.
+
+  If path exists and is a symlink, returns `{:ok, target}`, otherwise returns `{:error, reason}`.
+
+  For more details, see [`:file.read_link/1`](http://erlang.org/doc/man/file.html#read_link-1).
+
+  Typical error reasons are:
+  
+    * `:einval` - path is not a symbolic link
+    * `:enoent` - path does not exist
+    * `:enotsup` - symbolic links are not supported on the current platform
+  """
+  @spec read_link(Path.t) :: {:ok, binary} | {:error, posix}
+  def read_link(path) do
+    case path |> IO.chardata_to_string |> F.read_link do
+      {:ok, target} -> {:ok, target |> IO.chardata_to_string}
```

Comment:
The `|>` operator usage here is not necessary, let's use `IO.chardata_to_string(target)` instead.